### PR TITLE
[DO-NOT-MERGE] Drop support for Kubernetes v1.10

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Drop support for Kubernetes v1.10. 
Ref https://github.com/gardener/gardener/issues/1118#issuecomment-506496883

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```action user
:warning: Gardener does no longer support Kubernetes v1.10. You must upgrade all shoot cluster to at least Kubernetes v1.11 before upgrading Gardener. This also means that no seed clusters with versions prior to v1.11 are supported anymore. Please upgrade all seed clusters to at least v1.11 as well before upgrading Gardener cluster itself to v1.11.
```

### CODE REVIEW

File: pkg/apis/garden/types.go [hunk 1]
```diff
@@ -177,10 +177,6 @@ type OpenStackProfile struct {
 	// DNSServers is a list of IPs of DNS servers used while creating subnets.
 	// +optional
 	DNSServers []string
-	// DHCPDomain is the dhcp domain of the OpenStack system configured in nova.conf. Only meaningful for
-	// Kubernetes 1.10.1+. See https://github.com/kubernetes/kubernetes/pull/61890 for details.
-	// +optional
-	DHCPDomain *string
```
Comment (rfranzke) [inline pkg/apis/garden/types.go]: Don't remove the field entirely without a proper deprecation period. Mark it rather as "deprecated" and don't use it anymore anywhere.

File: pkg/apis/garden/v1beta1/types.go [hunk 2]
```diff
@@ -178,10 +178,6 @@ type OpenStackProfile struct {
 	// DNSServers is a list of IPs of DNS servers used while creating subnets.
 	// +optional
 	DNSServers []string `json:"dnsServers,omitempty"`
-	// DHCPDomain is the dhcp domain of the OpenStack system configured in nova.conf. Only meaningful for
-	// Kubernetes 1.10.1+. See https://github.com/kubernetes/kubernetes/pull/61890 for details.
-	// +optional
-	DHCPDomain *string `json:"dhcpDomain,omitempty"`
```
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go]: Don't remove the field entirely without a proper deprecation period. Mark it rather as "deprecated" and don't use it anymore anywhere.

**Review by rfranzke — CHANGES_REQUESTED** [review]
* Remove https://github.com/gardener/gardener/blob/master/pkg/apis/garden/types.go#L1296 and https://github.com/gardener/gardener/blob/master/pkg/apis/garden/v1beta1/types.go#L1297
* [This test](https://github.com/gardener/gardener/blob/master/pkg/apis/garden/validation/validation_test.go#L5103-L5112) can be simplified/merged into the previous test case now that we do not longer need to distinguish between versions < 1.11.
* Adapt https://github.com/gardener/gardener/blob/master/pkg/controllermanager/apis/config/types.go#L156-L157 and https://github.com/gardener/gardener/blob/master/pkg/controllermanager/apis/config/v1alpha1/types.go#L160-L161

**Comment by ialidzhikov** [comment]
Thank you for the review! Applied the suggestions.

**Comment by ialidzhikov** [comment]
Closing for now.

**Pull request closed without merging**